File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    answer = []
    arr = []
    for i in range(len(score)):
        if i <= k - 1:
            arr.append(score[i])
        else:
            if score[i] > min(arr):
                arr.pop(arr.index(min(arr)))
                arr.append(score[i])
        answer.append(min(arr))
    return answer

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.js
function solution(k, score) {
    let arr = [];
    let answer = [];
    for(let i=0; i<score.length; i++){
        if(arr.length <k){
            arr.push(score[i]);
        }
        else{
            arr.sort((a,b) => a-b);
            if(arr[0] < score[i]){
                arr[0] = score[i];
            }
        }
        
        answer.push(Math.min(...arr));
    }
    return answer;
}